/***************************************************************************
 *   This file is part of the CuteReport project                           *
 *   Copyright (C) 2014 by Ivan Volkov                                     *
 *   wulff007@gmail.com                                                    *
 *                                                                         *
 **                   GNU General Public License Usage                    **
 *                                                                         *
 *   This library is free software: you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation, either version 3 of the License, or     *
 *   (at your option) any later version.                                   *
 *   You should have received a copy of the GNU General Public License     *
 *   along with this program.  If not, see <http://www.gnu.org/licenses/>. *
 *                                                                         *
 **                  GNU Lesser General Public License                    **
 *                                                                         *
 *   This library is free software: you can redistribute it and/or modify  *
 *   it under the terms of the GNU Lesser General Public License as        *
 *   published by the Free Software Foundation, either version 3 of the    *
 *   License, or (at your option) any later version.                       *
 *   You should have received a copy of the GNU Lesser General Public      *
 *   License along with this library.                                      *
 *   If not, see <http://www.gnu.org/licenses/>.                           *
 *                                                                         *
 *   This library is distributed in the hope that it will be useful,       *
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
 *   GNU General Public License for more details.                          *
 ***************************************************************************/
#ifndef SQLSTORAGEHELPER_H
#define SQLSTORAGEHELPER_H

#include <QWidget>

#include "storageinterface.h"
#include "sql.h"

namespace Ui {
class SqlStorageHelper;
}

class SqlStorageHelper : public CuteReport::StorageHelperInterface
{
    Q_OBJECT

public:
    explicit SqlStorageHelper(StorageSql * storage, VisibleOptions visibleOptions = AllOptions);
    ~SqlStorageHelper();

public slots:
    virtual void load();
    virtual void save();

private:
    void connectObject();
    void disconnectObject();

    Ui::SqlStorageHelper *ui;
    StorageSql *m_storage;
};

#endif // SQLSTORAGEHELPER_H
